enforcer/src/keystate/keystate_ds_submit_task.c

```diff
@@ -48,7 +48,7 @@ keystate_ds_submit_task_perform(task_type *task)
 		KEY_DATA_DS_AT_PARENT_SUBMIT, KEY_DATA_DS_AT_PARENT_SUBMITTED,
 		(engine_type*)task->context);
 	task_cleanup(task);
-	flush_enforce_task((engine_type*)task->context, 0);
+	flush_enforce_task((engine_type*)task->context, time_now());
```

Comment:
@hodarh Yes you are correct. It is bad code. But it just so happens the context is the global engine struct. and there is no clean callback for the context on that particular task. Let's leave it for now and include it in the scheduler improvements.